In [2]:
import pandas as pd

### Data Pre-Processing

In [3]:
df = pd.read_csv(r"C:\Users\gjrah\Documents\Data Science\Pandas\Datasets\AB_NYC_2019.csv");
df.head(6)

id                                              name  host_id  \
0  2539                Clean & quiet apt home by the park     2787   
1  2595                             Skylit Midtown Castle     2845   
2  3647               THE VILLAGE OF HARLEM....NEW YORK !     4632   
3  3831                   Cozy Entire Floor of Brownstone     4869   
4  5022  Entire Apt: Spacious Studio/Loft by central park     7192   
5  5099         Large Cozy 1 BR Apartment In Midtown East     7322   

     host_name neighbourhood_group neighbourhood  latitude  longitude  \
0         John            Brooklyn    Kensington  40.64749  -73.97237   
1     Jennifer           Manhattan       Midtown  40.75362  -73.98377   
2    Elisabeth           Manhattan        Harlem  40.80902  -73.94190   
3  LisaRoxanne            Brooklyn  Clinton Hill  40.68514  -73.95976   
4        Laura           Manhattan   East Harlem  40.79851  -73.94399   
5        Chris           Manhattan   Murray Hill  40.74767  -73.97500   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0     Private room    149               1                  9  2018-10-19   
1  Entire home/apt    225               1                 45  2019-05-21   
2     Private room    150               3                  0         NaN   
3  Entire home/apt     89               1                270  2019-07-05   
4  Entire home/apt     80              10                  9  2018-11-19   
5  Entire home/apt    200               3                 74  2019-06-22   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2                NaN                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0  
5               0.59                               1               129

In [4]:
df = df.drop(['id','name','latitude','longitude','last_review','reviews_per_month','calculated_host_listings_count'],axis=1)
df

host_id      host_name neighbourhood_group       neighbourhood  \
0          2787           John            Brooklyn          Kensington   
1          2845       Jennifer           Manhattan             Midtown   
2          4632      Elisabeth           Manhattan              Harlem   
3          4869    LisaRoxanne            Brooklyn        Clinton Hill   
4          7192          Laura           Manhattan         East Harlem   
...         ...            ...                 ...                 ...   
48890   8232441        Sabrina            Brooklyn  Bedford-Stuyvesant   
48891   6570630        Marisol            Brooklyn            Bushwick   
48892  23492952  Ilgar & Aysel           Manhattan              Harlem   
48893  30985759            Taz           Manhattan      Hell's Kitchen   
48894  68119814     Christophe           Manhattan      Hell's Kitchen   

             room_type  price  minimum_nights  number_of_reviews  \
0         Private room    149               1                  9   
1      Entire home/apt    225               1                 45   
2         Private room    150               3                  0   
3      Entire home/apt     89               1                270   
4      Entire home/apt     80              10                  9   
...                ...    ...             ...                ...   
48890     Private room     70               2                  0   
48891     Private room     40               4                  0   
48892  Entire home/apt    115              10                  0   
48893      Shared room     55               1                  0   
48894     Private room     90               7                  0   

       availability_365  
0                   365  
1                   355  
2                   365  
3                   194  
4                     0  
...                 ...  
48890                 9  
48891                36  
48892                27  
48893                 2  
48894                23  

[48895 rows x 9 columns]

- Dropping the columns 'id','name','latitude','longitude','last_review','reviews_per_month','calculated_host_listings_count'

In [5]:
df = df.drop(['host_name','neighbourhood_group','neighbourhood'],axis=1);
df

host_id        room_type  price  minimum_nights  number_of_reviews  \
0          2787     Private room    149               1                  9   
1          2845  Entire home/apt    225               1                 45   
2          4632     Private room    150               3                  0   
3          4869  Entire home/apt     89               1                270   
4          7192  Entire home/apt     80              10                  9   
...         ...              ...    ...             ...                ...   
48890   8232441     Private room     70               2                  0   
48891   6570630     Private room     40               4                  0   
48892  23492952  Entire home/apt    115              10                  0   
48893  30985759      Shared room     55               1                  0   
48894  68119814     Private room     90               7                  0   

       availability_365  
0                   365  
1                   355  
2                   365  
3                   194  
4                     0  
...                 ...  
48890                 9  
48891                36  
48892                27  
48893                 2  
48894                23  

[48895 rows x 6 columns]

- Dropping the columns 'host_name','neighbourhood_group','neighbourhood'

In [6]:
#Information about data frame
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   host_id            48895 non-null  int64 
 1   room_type          48895 non-null  object
 2   price              48895 non-null  int64 
 3   minimum_nights     48895 non-null  int64 
 4   number_of_reviews  48895 non-null  int64 
 5   availability_365   48895 non-null  int64 
dtypes: int64(5), object(1)
memory usage: 2.2+ MB


##### We have 5 colums in total
- host_id
- room_type
- price,minimum_nights
- number_of_reviews
- availability_365

In [7]:
#Check isnull values
df.isnull().sum()

host_id              0
room_type            0
price                0
minimum_nights       0
number_of_reviews    0
availability_365     0
dtype: int64

##### We have no null values

In [8]:
lower_limit = df.price.quantile(0.5)
upper_limit = df.price.quantile(0.95)

print(f"Lower Limit: {lower_limit}")
print(f"Upper Limit: {upper_limit}")

Lower Limit: 106.0
Upper Limit: 355.0


In [9]:
df_price = df[df.price > 355]
df_price.price.sample(5)

42688    375
41987    425
46276    450
42390    400
45886    995
Name: price, dtype: int64

- We can see we have higher values than 355
- Which are outliners
- Outliners are the values that are more than the average value of the data present (not the actual defination)
- I have decided to keep the lower limit as 5%
- Upper Limit as 95%
- When lower limit is 106
- Higher limit is 355
- 99% could be 360
- Anything other than 360 is outliner

In [10]:
df = df[(df.price >= lower_limit) & (df.price<= upper_limit)]
df

host_id        room_type  price  minimum_nights  number_of_reviews  \
0           2787     Private room    149               1                  9   
1           2845  Entire home/apt    225               1                 45   
2           4632     Private room    150               3                  0   
5           7322  Entire home/apt    200               3                 74   
9           7549  Entire home/apt    150               1                160   
...          ...              ...    ...             ...                ...   
48884  274307600     Private room    190               7                  0   
48886  274311461  Entire home/apt    200               6                  0   
48887  208514239  Entire home/apt    170               1                  0   
48888  274321313     Private room    125               4                  0   
48892   23492952  Entire home/apt    115              10                  0   

       availability_365  
0                   365  
1                   355  
2                   365  
5                   129  
9                   188  
...                 ...  
48884               341  
48886               176  
48887               365  
48888                31  
48892                27  

[22017 rows x 6 columns]

In [12]:
max_value = df.price.max()
min_value = df.price.min()


In [13]:
#Dropping the un-necessary cols
df = df.drop(['host_id','number_of_reviews'],axis=1)

In [14]:
#All the values that holds the max_value for price
df[df["price"]==max_value].shape

(10, 4)

In [15]:
#All values that holds the min_value for the price
df[df['price'] == min_value].shape

(35, 4)